apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/trigger_policy/queries.ex

```diff
@@ -18,36 +18,30 @@
 
 defmodule Astarte.DataUpdaterPlant.TriggerPolicy.Queries do
   require Logger
-  alias Astarte.Core.CQLUtils
-  alias Astarte.DataUpdaterPlant.Config
+
+  import Ecto.Query
+
+  alias Astarte.DataUpdaterPlant.Repo
+  alias Astarte.DataUpdaterPlant.DataUpdater.KvStore
+  alias Astarte.DataUpdaterPlant.DataUpdater.Realm
 
   def retrieve_policy_name(realm_name, trigger_id) do
+    keyspace_name = Realm.keyspace_name(realm_name)
+
     trigger_id =
       trigger_id
       |> :uuid.uuid_to_string()
       |> to_string()
 
-    keyspace_name =
-      CQLUtils.realm_name_to_keyspace_name(realm_name, Config.astarte_instance_id!())
-
-    Xandra.Cluster.run(:xandra, &do_retrieve_policy_name(&1, keyspace_name, trigger_id))
-  end
-
-  defp do_retrieve_policy_name(conn, keyspace_name, trigger_id) do
-    retrieve_statement =
-      "SELECT value FROM #{keyspace_name}.kv_store WHERE group='trigger_to_policy' AND key=:trigger_id;"
+    query =
+      from kvstore in KvStore,
+        prefix: ^keyspace_name,
+        where: kvstore.group == "trigger_to_policy" and kvstore.key == ^trigger_id,
+        select: kvstore.value
 
-    with {:ok, prepared} <- Xandra.prepare(conn, retrieve_statement),
-         {:ok, %Xandra.Page{} = page} <-
-           Xandra.execute(conn, prepared, %{"trigger_id" => trigger_id}) do
-      case Enum.to_list(page) do
-        [%{"value" => policy_name}] -> {:ok, policy_name}
-        [] -> {:error, :policy_not_found}
-      end
-    else
-      {:error, error} ->
-        Logger.warning("Database error #{inspect(error)}", tag: "retrieve_policy_name_db_error")
-        {:error, error}
+    case Repo.all(query) do
+      [%{"value" => policy_name}] -> {:ok, policy_name}
+      [] -> {:error, :policy_not_found}
```

Comment:
```suggestion
    case Repo.one(query) do
      nil -> {:error, :policy_not_found}
      value -> {:ok, value}
```

unfortunately it's not obvious from the tests as it always returns `{:error, :policy_not_found}`, but when you `select:` a single field from the capture it returns just that value and not the whole struct (it would have had to be `value:` (the atom) even if it was the whole struct)

Reply:
Ok! I understood that it was better to use `Repo.all` (it lifts less times)